master/internal/api_workspace.go

```diff
@@ -519,12 +558,31 @@ func (a *apiServer) deleteWorkspaceNamespaceBinding(ctx context.Context,
 			if err != nil {
 				return err
 			}
+			deletedClusters := []string{}
 			for _, v := range deletedBindings {
+				deletedClusters = append(deletedClusters, v.ClusterName)
 				err = a.m.rm.RemoveEmptyNamespace(v.Namespace, v.ClusterName)
 				if err != nil {
 					return err
 				}
 			}
+			if len(deletedClusters) < len(clusterNames) {
```

Comment:
If we only use `deletedClusters` here, maybe we remove that array and do 
`if len(deletedBindings) < len(clusterNames) {` ?

Reply:
Can we add a comment here that's like 
`// Since bindings to the default namespace are not stored in the db, we may delete less bindings in the database than specified in the request`

Reply:
done